static/index.js

```diff
@@ -39,17 +43,58 @@ document.addEventListener("DOMContentLoaded", event => {
     event: "doubletap",
     taps: 2
   });
-  const hammer = new Hammer.Manager(document.getElementById("touchpad"));
+
   hammer.add([doubleTap, singleTap]);
-  doubleTap.recognizeWith(singleTap);
-  doubleTap.requireFailure(singleTap);
-  singleTap.requireFailure(doubleTap);
   hammer.on("tap", ev => {
     // left mouse button click
     ws.send(JSON.stringify({ type: "tap" }));
   });
   hammer.on("doubletap", ev => {
-    // right mouse button click
+    // right mouse button click 
     ws.send(JSON.stringify({ type: "doubletap" }));
   });
+  doubleTap.recognizeWith(singleTap);
+  doubleTap.requireFailure(singleTap);
+  singleTap.requireFailure(doubleTap);
+
+  var scrollyup = new Hammer.Pan({
+    event: "vscrollup",
+    threshhold: 50,
+    direction: Hammer.DIRECTION_UP
+  });
+  var scrollydown = new Hammer.Pan({
+    event: "vscrolldown",
+    threshhold: 50,
+    direction: Hammer.DIRECTION_DOWN
+  });
+  var scrollxright= new Hammer.Pan({
+    event: "hscrollright",
+    threshhold: 50,
+    direction: Hammer.DIRECTION_RIGHT
+  });
+  var scrollxleft = new Hammer.Pan({
+    event: "hscrollleft",
+    threshhold: 50,
+    direction: Hammer.DIRECTION_LEFT
+  });
+
+
+  vscrollHammer.add([scrollyup,scrollydown]);
+  hscrollHammer.add([scrollxright, scrollxleft]);
+  vscrollHammer.on("vscrollup", ev => { 
+    // scroll up
+    ws.send(JSON.stringify({ type: "scrollyup"}));
+  });
+  vscrollHammer.on("vscrolldown", ev => {
+    // scroll down
+    ws.send(JSON.stringify({ type: "scrollydown"}));
+  });
+  hscrollHammer.on("hscrollright", ev => {
+    // scroll right 
+    ws.send(JSON.stringify({ type: "scrollxright" }));
+  });
+  hscrollHammer.on("hscrollleft", ev => {
```

Comment:
Somehow horizontal scrolling is not working for me. Is it working for you? Overall scrolling feels a little jittery. If you can tweak some params to fix that it'll be great, or we can also track this separately.

Reply:
According to pyautogui's documentation, the hscroll() performs horizontal scrolling on OS X and Linux platforms. I'm not exactly sure how to fix this as I've looked up many websites to no avail. Maybe I'm not looking hard enough. 

![mouseScroll](https://user-images.githubusercontent.com/42780124/156150665-45f2c722-6c79-4408-9fd5-17d5e8385616.png)
 
Also, I did some testing just now. I connected to my computer with my Android phone and iPad and noticed that the movements through the phone were much laggier compared to the iPad.

Here is a video: iPad first, Android phone second.

https://user-images.githubusercontent.com/42780124/156151183-ed1fbb69-ecb1-440e-bd1a-18366bd52767.mp4

Reply:
Got it. There are some platform limitations in `pyautogui` and sadly we don't have any other option but to fix it there and consume it here. We have a hack in this repository already for modifying the source code in place for OSX volume bindings. Let's just add a disclaimer to Readme that `hscroll` is only supported on OSX/Linux.

I am also facing lags when I use my Android device. It might be something to do with the browser and the connection quality. This is something we can optimize later.